File: 2138-divide-a-string-into-groups-of-size-k/2138-divide-a-string-into-groups-of-size-k.java
class Solution {
    public String[] divideString(String s, int k, char fill) {
        int n=s.length();
        int t=n/k;
        ArrayList<String> str = new ArrayList<>();

        int r=n%k;
        int i=0;
        while(t!=0){
            String temp=s.substring(i,i+k);
            str.add(temp);
            i+=k;
            t--;
        }
        if (r!=0){
            
            t=k-r;
            String temp=s.substring(i);
            while(t!=0){
                temp+=fill;
                t--;
            }
            str.add(temp);
        }
        String[] res=str.toArray(new String[0]);
        return res;
    }
}

File: 2138-divide-a-string-into-groups-of-size-k/2138-divide-a-string-into-groups-of-size-k.cpp
class Solution {
public:
    vector<string> divideString(string s, int k, char fill) {
        vector<string>res;
        int n=s.size();
        int t=n/k;
        int r=n%k;
        int i=0;
        while(t--){
            string temp=s.substr(i,k);
            res.push_back(temp);
            i+=k;
        }
        if (r!=0){
            
            t=k-r;
            string temp=s.substr(i);
            while(t--){
                temp+=fill;
            }
            res.push_back(temp);
        }
        return res;
    }
};